src/parser/parser.yy

```diff
@@ -730,6 +738,40 @@ when_then_list
     }
     ;
 
+
+list_comprehension_expression
+    : L_BRACKET expression KW_IN expression KW_WHERE expression R_BRACKET {
+        if ($2->kind() != Expression::Kind::kLabel) {
+            throw nebula::GraphParser::syntax_error(@2, "Must be LabelExpression:");
```

Comment:
Please be more specific on the error report. Like _**The loop variable must be a label in list comprehension**_

Reply:
Thanks for your advice